examples/platforms/eagle/radio.c

```diff
@@ -0,0 +1,791 @@
+                    sAckFrame.mPsdu[2] = otMacFrameGetSequence(rx_frame_ptr);
+
+                    sAckFrame.mChannel = rx_frame_ptr->mChannel;
+
+                    // Transmit
+                    setupTransmit(&sAckFrame);
+                    rf_set_txmode();
+                    rf_tx_pkt(tx_buffer);
+                    // rf_start_stx(tx_buffer,0);
+                }
+
+                // push the frame
+                empty--;
+                w_ptr = (w_ptr + 1) % RX_FRAME_SLOT_NUM;
+                full++;
+            }
+        }
+
+    exit:
+        dma_chn_en(DMA1);
+    }
+    else if (rf_get_irq_status(FLD_RF_IRQ_TX))
+    {
+        rf_clr_irq_status(FLD_RF_IRQ_TX);
+
+        if ((tx_busy == 1) && (sState == OT_RADIO_STATE_TRANSMIT))
+        {
+            tx_busy = 0;
+        }
+        // set to rx mode
+        rf_set_rxmode();
+    }
+
+    // plic_interrupt_complete(IRQ15_ZB_RT);
+}
+
+void eagleRadioProcess(otInstance *aInstance)
+{
+    otRadioFrame *ptr;
+    otRadioFrame *rx_frame_ptr;
+    int           i;
+
+    ptr = (otRadioFrame *)rx_frame;
+    util_disable_rf_irq();
+    if (full > 0)
+    {
+        full--;
+        rx_frame_ptr  = (otRadioFrame *)rx_frame_slots[r_ptr];
+        ptr->mChannel = rx_frame_ptr->mChannel;
+        ptr->mLength  = rx_frame_ptr->mLength;
+        for (i = 0; i < ptr->mLength - 2; i++)
+        {
+            ptr->mPsdu[i] = rx_frame_ptr->mPsdu[i];
+        }
+
+        ptr->mInfo.mRxInfo.mRssi                  = rx_frame_ptr->mInfo.mRxInfo.mRssi;
+        ptr->mInfo.mRxInfo.mLqi                   = rx_frame_ptr->mInfo.mRxInfo.mLqi;
+        ptr->mInfo.mRxInfo.mAckedWithFramePending = rx_frame_ptr->mInfo.mRxInfo.mAckedWithFramePending;
+        r_ptr                                     = (r_ptr + 1) % RX_FRAME_SLOT_NUM;
+        empty++;
+        util_enable_rf_irq();
+    }
+    else
+    {
+        util_enable_rf_irq();
+        ptr = NULL;
+    }
+
+    if ((sState == OT_RADIO_STATE_RECEIVE) || (sState == OT_RADIO_STATE_TRANSMIT))
+    {
+        if ((ptr != NULL) && (ptr->mLength > IEEE802154_ACK_LENGTH))
+        {
+            otPlatRadioReceiveDone(aInstance, ptr, sReceiveError);
+        }
+    }
+
+    if (sState == OT_RADIO_STATE_TRANSMIT)
+    {
+        if (sTransmitError != OT_ERROR_NONE ||
+            (((sTransmitFrame.mPsdu[0] & IEEE802154_ACK_REQUEST) == 0) && (tx_busy == 0)))
+        {
+            sState = OT_RADIO_STATE_RECEIVE;
+            otPlatRadioTxDone(aInstance, &sTransmitFrame, NULL, sTransmitError);
+        }
+        else if ((ptr != NULL) && ptr->mLength == IEEE802154_ACK_LENGTH &&
+                 (ptr->mPsdu[0] & IEEE802154_FRAME_TYPE_MASK) == IEEE802154_FRAME_TYPE_ACK &&
+                 (ptr->mPsdu[IEEE802154_DSN_OFFSET] == sTransmitFrame.mPsdu[IEEE802154_DSN_OFFSET]))
+        {
+            sState = OT_RADIO_STATE_RECEIVE;
+            otPlatRadioTxDone(aInstance, &sTransmitFrame, ptr, sTransmitError);
```

Comment:
It seems the driver doesn't process the diag  APIs, such as `otPlatDiagRadioTransmitDone`. Could we add the support for the diag?